language/vm/src/views.rs

```diff
@@ -509,12 +509,63 @@ impl<'a, T: ModuleAccess> LocalsSignatureView<'a, T> {
     }
 }
 
+/// Returns the kind of the signature token in the given context (module, function/struct).
+/// The context is needed to determine the kinds of structs & type variables.
+// TODO: rework views to make the interfaces here cleaner.
+pub fn kind_impl(context: (&[StructHandle], &[Kind]), ty: &SignatureToken) -> Kind {
```

Comment:
May refactor this/rename this/move this around.

Reply:
Maybe just call this `kind`?